am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -246,60 +238,40 @@ public FilteredResourceEnvelope filterResource(@NotBlank String userId,
             .resource(Resource.builder()
                 .id(resource.getId())
                 .definition(resource.getDefinition())
-                .data(filteredJson)
+                .data(filteredData)
                 .build())
             .userSecurityClassification(userSecurityClassification)
             .access(AccessEnvelope.builder()
-                .permissions(attributePermissions)
+                .permissions(visibleAttributePermissions)
                 .accessType(accessType)
                 .build())
             .relationships(relationships)
             .build();
     }
 
-    /**
-    * Filter data with higher security classification.
-    *
-    * @param attributePermissions set
-    * @param filteredJson JsonNode
-    * @return JsonNode
-    */
-    private JsonNode filterDataWithHigherSecurityClassification(Map<JsonPointer, Set<Permission>> attributePermissions,
-                                                                JsonNode filteredJson) {
-
-        List<JsonPointer> visibleAttributes = attributePermissions.entrySet().stream()
-            .map(s -> s.getKey()).collect(toList());
-        filterService.retainFieldsWithVisibleSecurityClassifications(filteredJson, visibleAttributes);
-        return filteredJson;
-    }
+    private Map<JsonPointer, Set<Permission>> filterAttributePermissionsBySecurityClassification(
```

Comment:
Please add java docs. May be some comments around logic as it has been refactored and holding main logic

Reply:
👍 added some inline comments for main logic